Fix example app

kevc opened the pull request:
@pardom 

Some fixes for `plumb-example`.
* Fix building with Android Studio
* Fix crash on rotate to landscape
* Fix subscription management

pardom-zz commented on compiler/src/test/kotlin/plumb/internal/PlumberWriterTest.kt:
```diff
@@ -46,10 +46,11 @@ class PlumberWriterTest {
             "",
             "	@Override",
             "	public void plumb(PlumbedClassA plumbed, PlumbedClassA.PlumbedViewModelA plumbedTo) {",
+            "       subscriptions = new CompositeSubscription();",
```
formatting

pardom-zz commented on build.gradle:
```diff
@@ -6,6 +6,7 @@ buildscript {
     }
     dependencies {
         classpath 'com.android.tools.build:gradle:2.0.0-beta6'
```
These should go in the Android project's Gradle config as they aren't used anywhere else.

kevc commented:
Good call on the subscription check. Added that, addressed formatting, gradle structure, and added the "DO NOT MODIFY" comment to the top of all generated files. :v:
